s3_management/manage.py

```diff
@@ -166,10 +165,12 @@ def nightly_packages_to_show(self: S3IndexType) -> Set[str]:
                 to_hide.add(obj)
             else:
                 packages[package_name] += 1
-        return set(self.objects).difference({
-            obj for obj in self.objects
-            if self.normalize_package_version(obj) in to_hide
-        })
+        nightly_packages = {}
+        for obj, checksum in self.objects.items():
+            normalized_package_version = self.normalize_package_version(obj)
+            if not normalized_package_version in to_hide:
+                nightly_packages[normalized_package_version] = checksum
+        return nightly_packages
```

Comment:
Maybe simplify to a comprehension?

```suggestion
        return {
            s3_key: checksum
            for s3_key, checksum in self.objects.items()
            if self.normalize_package_version(s3_key) not in to_hide
        }
```

Reply:
Wouldn't it need to be this which would duplicate the call to `normalize_package_version`?
```
        return {
            self.normalize_package_version(s3_key): checksum
            for s3_key, checksum in self.objects.items()
            if self.normalize_package_version(s3_key) not in to_hide
        }
```

Reply:
As best I can tell, the old code did `set(self.objects)` meaning it wasn't using the return of `normalize_package_version` as the key.